zcash_primitives/src/transparent.rs

```diff
@@ -42,4 +43,54 @@ impl ExternalPubKey {
     pub fn public_key(&self) -> &secp256k1::key::PublicKey {
         &self.0.public_key
     }
+
+    /// Returns the chain code component of this public key.
+    pub fn chain_code(&self) -> &[u8] {
+        &self.0.chain_code
+    }
+
+    /// Derives the internal ovk and external ovk corresponding to this
+    /// transparent fvk. As specified in [ZIP 316][transparent-ovk].
+    ///
+    /// [transparent-ovk]: https://zips.z.cash/zip-0316#deriving-internal-keys
+    fn ovk_for_shielding(&self) -> (InternalOvk, ExternalOvk) {
```

Comment:
This method name confused me, as the OVK for shielding is just the internal part. I might just call this `derive_ovks` instead.